sound/soc/bcm/hifiberry_dacplusadcpro.c

```diff
@@ -414,6 +416,13 @@ static int snd_rpi_hifiberry_dacplusadcpro_hw_params(
 		return ret;
 	ret = snd_soc_dai_set_tdm_slot(rtd->codec_dais[1], 0x03, 0x03,
 		channels, width);
+	if (ret)
+		return ret;
+
+	if (snd_rpi_hifiberry_is_dacpro) {
```

Comment:
You can turn the nested if into an && and lose the braces. Otherwise I'm OK with this patch.